Added API calls for system save/load dialogs for use from scripts.

ibbo opened the pull request:
I plan to use this for the save/load table state functionality in the Lord of the Rings plugin. I wanted to make it possible to use the system dialogs to choose the location to save the table state and the file to load it from.

I went for the simplest solution of exposing the system SaveFileDialog and OpenFileDialog via the script without adding another Control to Octgn/Scripting/Controls, which I tried initially, but ultimately wasn't necessary for my use case.

kellyelton commented on octgnFX/Octgn/Scripting/Versions/Script_3_1_0_2.cs:
```diff
@@ -1020,6 +1020,43 @@ public string AskString(string question, string defaultValue)
+                var filePath = string.Empty;
+                using (OpenFileDialog openFileDialog = new OpenFileDialog())
+                {
+                    openFileDialog.InitialDirectory = "c:\\";
```
I think we should make the default directory the OctgnDataDirectory, Users Documents, or the Users desktop. Having `C` drive isn't a guarantee on Windows (it can be changed).

kellyelton commented on octgnFX/Octgn/Scripting/Versions/Script_3_1_0_2.cs:
```diff
@@ -1020,6 +1020,43 @@ public string AskString(string question, string defaultValue)
+                {
+                    openFileDialog.InitialDirectory = "c:\\";
+                    openFileDialog.Filter = fileFilter;
+                    openFileDialog.FilterIndex = 2;
```
We should either give the caller the option to specify a FilterIndex, or just remove this line so it uses the default.

ibbo commented:
I've added a new commit addressing your comments, and some of Brine's comments from a discussion on Discord. I'm not sure what the process is for notifying you of these changes since this is my first Github pull request. So I'm using this comment as a notification, sorry if that's not the right way to do it.
